cookies/views/resource.py

```diff
@@ -673,6 +741,51 @@ def resource_content(request, resource_id):
     return HttpResponse('Nope')    # TODO: say something more informative!
 
 
+@auth.authorization_required(ResourceAuthorization.VIEW, _get_resource_by_id)
+def resource_text_content(request, resource_id):
+    """
+    Serve up the raw text content associated with a :class:`.Resource`\.
+
+    This is not the most efficient way to serve files, but we need some kind of
+    security layer here for non-public content.
+    """
+
+    parent_resource = _get_resource_by_id(request, resource_id)
+    content_relations = parent_resource.content.filter(is_deleted=False, content_type='text/plain')
+    if not content_relations:
+        return HttpResponse('No text content found for resource \'%s\'.' % parent_resource.name)
+
+    content_relation = content_relations[0]
+    resource = content_relation.content_resource
+
+    content_type = resource.content_type
+    if resource.file:
+        try:
+            with open(resource.file.path, 'rb') as f:
+                return HttpResponse(f.read(), content_type=content_type)
+        except IOError:
+            return HttpResponse('Couldn\'t read text content for resource \'%s\'.' % parent_resource.name)
+    elif resource.location:
+        cache = caches['remote_content']
+
+        remote = get_remote(resource.external_source, resource.created_by)
+        target = resource.location
+        if resource.external_source == Resource.GILES:
+            target = _add_parameter(target, 'dw', 300)
```

Comment:
the dw paramter should only be required when requesting images. Is this really necessary here?

Reply:
or better, this should only be required when content type is image